ecl/eclccserver/eclccserver.cpp

```diff
@@ -464,7 +464,30 @@ class EclccCompileThread : implements IPooledThread, implements IErrorReporter,
 #ifdef _CONTAINERIZED
         if (!globals->getPropBool("@useChildProcesses", false) && !globals->hasProp("@workunit"))
         {
-            runK8sJob("eclccserver", wuid, wuid, globals->getPropBool("@deleteJobs", true));
+            try
+            {
+                runK8sJob("compile", wuid, wuid, globals->getPropBool("@deleteJobs", true));
+            }
+            catch (IException *E)
+            {
+                Owned<IWorkUnitFactory> factory = getWorkUnitFactory();
+                workunit.setown(factory->updateWorkUnit(wuid.get()));
+                if (workunit)
+                {
+                    if (workunit->aborting() || workunit->getState()==WUStateAborted)
+                        workunit->setState(WUStateAborted);
```

Comment:
not sure if causes a transaction or not, but no reason to call setState(WUStateAborted), if state is already WUStateAborted?

Reply:
Just copying the style of code from elsewhere in this function - and it's not going to be speed critical. The code is clearer like this I think.

Reply:
I agree with Jake, it doesn't make much sense to set the state to a value it already has.  I wonder if this code meant to say
   workunit->getState()==WUStateAborting
It looks like the code has been there since the original move from svn.

Reply:
I think the code is doing what it intended - the point of including the ==aborted test is to avoid taking the "else" arm.